### DISCUSSION TITLE
Implementation of existing cryptographic algorithms

### DISCUSSION THREAD

**Discussion by ranjithravitk** [openssl/openssl#23962] [Q&A] [answered]
Hi all, I would like to implement an existing cryptographic algorithm among Elgamal, salsa20, and Twofish which are not already implemented in OpenSSL. Is there a reason for not implementing it? I am looking for some input from the community, after I understand it I will create the pull request for it. I appreciate any help you can provide. I am looking into this as a part of my MSc dissertation.

**Comment by mattcaswell** [comment, marked as answer]
If you want to implement these I would recommend doing it as a third party provider (i.e. as a plug-in to OpenSSL).

To learn more about providers you might start here:

https://www.youtube.com/playlist?list=PLCCJVQZzhxHZe-dc5OS076e7RHkv_tOx6

**Reply by ranjithravitk** [reply]
Hi @mattcaswell , Is there a way to find out if Elgamal, salsa20, and Twofish are already implemented by someone else as a third-party provider?

**Reply by t8m** [reply]
This is a good starting point to look for third party providers: https://github.com/provider-corner

**Reply by ranjithravitk** [reply]
If additional algorithms were implemented as third-party providers, would that be included in the OpenSSL library after the review? Because I saw a folder named as provider in the repository. So, I was wondering if that is the case. Thanks in advance!!
